mapper/orm/src/main/java/org/hibernate/search/mapper/orm/session/SearchSessionWritePlan.java

```diff
@@ -6,16 +6,27 @@
  */
 package org.hibernate.search.mapper.orm.session;
 
-// TODO HSEARCH-3069 Also document what happens when a transaction is *not* active
 /**
  * An interface for writing to indexes in the context of an ORM Session.
  * <p>
- * When a transaction is active, operations are queued internally,
- * and are only applied on transaction commit.
+ * This class is stateful: it queues operations internally to apply them at a later time.
  * <p>
- * {@link #process()} and {@link #execute()} allow to control explicitly
- * how operations are applied, without relying on transaction commits,
- * which can be useful when indexing large amounts of data in batches.
+ * When {@link #process()} is called,
+ * or when {@link org.hibernate.search.mapper.orm.cfg.HibernateOrmMapperSettings#AUTOMATIC_INDEXING_STRATEGY automatic indexing is enabled}
+ * and a Hibernate ORM Session {@code flush()} happens,
+ * the entities will be processed and index documents will be built
+ * and stored in an internal buffer.
+ * <p>
+ * When {@link #execute()} is called,
+ * or when {@link org.hibernate.search.mapper.orm.cfg.HibernateOrmMapperSettings#AUTOMATIC_INDEXING_STRATEGY automatic indexing is enabled}
```

Comment:
Same here. I don't think strategy really matters. I would write that the execution of the prepared works will be triggered by a commit or by an out of transaction flush.

Reply:
Same answer.